platform/security/filter/security-filter-login/src/main/java/org/codice/ddf/security/filter/login/LoginFilter.java

```diff
@@ -198,28 +184,27 @@ public void doFilter(
         });
   }
 
-  private Object resolveReference(BaseAuthenticationToken token, HttpServletRequest httpRequest) {
-    if (token.isReference()) {
-      LOGGER.trace("Converting reference to assertion");
-      Object sessionTokenHolder =
-          httpRequest.getSession(false).getAttribute(SecurityConstants.SECURITY_TOKEN_KEY);
-      if (LOGGER.isTraceEnabled()) {
-        LOGGER.trace(
-            "Http Session assertion - class: {}  loader: {}",
-            sessionTokenHolder.getClass().getName(),
-            sessionTokenHolder.getClass().getClassLoader());
-        LOGGER.trace(
-            "SecurityToken class: {}  loader: {}",
-            SecurityToken.class.getName(),
-            SecurityToken.class.getClassLoader());
-      }
-      try {
-        return ((SecurityTokenHolder) sessionTokenHolder).getPrincipals();
-      } catch (ClassCastException e) {
-        httpRequest.getSession(false).invalidate();
-      }
-    }
-    return null;
+  // temporary method
+  private BaseAuthenticationToken checkSessionTokenExpiration(BaseAuthenticationToken token) {
+    //    if (token instanceof SessionToken) {
+    //      Collection<SecurityAssertion> securityAssertions =
+    //          ((PrincipalCollection) token.getCredentials()).byType(SecurityAssertion.class);
+    //      SecurityAssertionSaml securityAssertionSaml =
+    //          (SecurityAssertionSaml) securityAssertions
+    //              .stream()
+    //              .filter(sa -> SecurityAssertionSaml.SAML2_TOKEN_TYPE.equals(sa.getTokenType()))
+    //              .findFirst()
+    //              .orElse(null);
+    //      if (securityAssertionSaml != null) {
+    //        SAMLAuthenticationToken authToken =
+    //            new SAMLAuthenticationToken(
+    //                null, (PrincipalCollection) token.getCredentials(), token.getIpAddress());
+    //        authToken.setAllowGuest(token.getAllowGuest());
+    //        return authToken;
+    //      }
+    //    }
+    //    return token;
+    return token;
```

Comment:
📘 Still figuring out what to do with this method.

Reply:
I'm thinking that we might just make a new interface for this. SessionExpirationHandler, and then we can have a SAML implementation that does what is in this method. We don't need to do anything for any other tokens, but maybe we will in the future? Who knows.